ui/pages/confirmations/confirmation/confirmation.js

```diff
@@ -427,19 +430,23 @@ export default function ConfirmationPage({
   const handleSubmit = async () => {
     setLoading(true);
 
-    if (
-      pendingConfirmation?.requestData?.fromNetworkConfiguration?.chainId &&
-      pendingConfirmation?.requestData?.toNetworkConfiguration?.chainId
-    ) {
+    const fromNetwork =
+      pendingConfirmation?.requestData?.fromNetworkConfiguration?.chainId ||
+      chainId;
+
+    const toNetwork =
+      pendingConfirmation?.requestData?.toNetworkConfiguration?.chainId ||
+      pendingConfirmation?.requestData?.chainId ||
+      pendingConfirmation?.chainID;
```

Comment:
We should document why there are so many possible properties we can check.  I'd love to know why this can be found in (possibly) three places.

Reply:
I'm looking at the reason for this behaviour

Reply:
Swap & send: the **to** network is different from the **from** network

Dapp initiated transaction: the dapp request can be on a different chain than the globally selected chain

Otherwise, fall back to global chain

(This is my understanding)